packages/utils/plugins-helpers/src/federation.ts

```diff
@@ -345,19 +382,15 @@ export class ApolloFederation {
     return this.meta;
   }
 
-  private isExternalAndNotProvided(fieldNode: FieldDefinitionNode, objectType: GraphQLObjectType): boolean {
-    return this.isExternal(fieldNode) && !this.hasProvides(objectType, fieldNode);
-  }
-
-  private isExternal(node: FieldDefinitionNode): boolean {
+  private isExternal(node: FieldDefinitionNode | ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode): boolean {
     return getDirectivesByName('external', node).length > 0;
   }
 
-  private hasProvides(objectType: ObjectTypeDefinitionNode | GraphQLObjectType, node: FieldDefinitionNode): boolean {
-    const fields = this.providesMap[isObjectType(objectType) ? objectType.name : objectType.name.value];
+  private hasProvides(node: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode, fieldName: string): boolean {
```

Comment:
Note for the future: I find allowing both types (e.g. `GraphQLObjectType`) and nodes (e.g. `ObjectTypeDefinitionNode`) can get confusing very quickly. 

Maybe we should just only accept nodes in the future?